spec/support/fake_ansible_repo.rb

```diff
@@ -0,0 +1,357 @@
+# for a playbook:
+#
+#     - name: filename.yml
+#       hosts: all
+#       tasks:
+#         - name: filename.yml Message
+#           debug:
+#             msg: "Hello World! (from filename.yml)"
+#
+# This is mostly to allow for testing playbook detecting code.  To the end,
+# there are also "reserved" filenames that you can use to generate alternative
+# dummy content that is not in a playbook format.
+#
+#
+# === requirements.yml
+#
+# Regardless of where you put this file, if found, it will generate a sample
+# `ansible-galaxy` requirements file.
+#
+#     - src: yatesr.timezone
+#     - src: https://github.com/bennojoy/nginx
+#     - src: https://github.com/bennojoy/nginx
+#       version: master
+#         name: nginx_role
+#
+#
+# === roles/**/meta/main.yml
+#
+# Will generate a "Role Dependencies" yaml definition:
+#
+#     ---
+#     dependencies:
+#       - role: common
+#         vars:
+#           some_parameter: 3
+#       - role: apache
+#         vars:
+#           apache_port: 80
+#
+#
+# === extra_vars.yml, roles/**/vars/*.yml, and roles/**/defaults/*.yml
+#
+# Generates sample variable files
+#
+#     ---
+#     var_1: foo
+#     var_2: bar
+#     var_3: baz
+#
+
+require "rugged"
+
+require "yaml"
+require "pathname"
+require "fileutils"
+
+module Spec
+  module Support
+    class FakeAnsibleRepo
+
+      # Content format/borrowed from the ansible documentation
+      #
+      #   https://docs.ansible.com/ansible/latest/reference_appendices/galaxy.html#installing-multiple-roles-from-a-file
+      REQUIREMENTS = <<~REQUIREMENTS
+        - src: yatesr.timezone
+        - src: https://github.com/bennojoy/nginx
+        - src: https://github.com/bennojoy/nginx
+          version: master
+            name: nginx_role
+      REQUIREMENTS
+
+      # Content format/borrowed from the ansible documentation
+      #
+      #   https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-dependencies
+      META_DATA = <<~ROLE_DEPS
+        ---
+        dependencies:
+          - role: common
+            vars:
+              some_parameter: 3
+          - role: apache
+            vars:
+              apache_port: 80
+      ROLE_DEPS
+
+      VAR_DATA = <<~VARS
+        ---
+        var_1: foo
+        var_2: bar
+        var_3: baz
+      VARS
+
+      attr_reader :file_struct, :repo_path, :repo, :index
+
+      def self.generate repo_path, file_struct
+        new(repo_path, file_struct).generate
+      end
+
+      def initialize(repo_path, file_struct)
+        @repo_path   = Pathname.new(repo_path)
+        @name        = @repo_path.basename
+        @file_struct = parse_file_struct(file_struct)
+      end
+
+      def generate
+        FileUtils.mkdir_p(repo_path)
+        build_files_at(repo_path, file_struct)
+
+        git_init
+        git_add_all
+        git_commit_initial
+      end
+
+      # Create a new branch (don't checkout)
+      #
+      #   $ git branch other_branch
+      #
```

Comment:
This comment feels obvious and thus not needed...reading the method name I didn't even consider that it might be a checkout.

Reply:
Was copy pasted from what I had in the previous spec.